utils/src/main/java/com/cloud/utils/server/ServerProperties.java

```diff
@@ -0,0 +1,82 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.utils.server;
+
+import com.cloud.utils.PropertiesUtil;
+import com.cloud.utils.crypt.EncryptionSecretKeyChecker;
+import org.apache.commons.io.IOUtils;
+import org.apache.log4j.Logger;
+import org.jasypt.encryption.pbe.StandardPBEStringEncryptor;
+import org.jasypt.properties.EncryptableProperties;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class ServerProperties {
+    private static final Logger log = Logger.getLogger(ServerProperties.class);
+
+    private static Properties properties = new Properties();
+    private static boolean loaded = false;
+
+    public synchronized static Properties getServerProperties() {
+        if (!loaded) {
+            Properties serverProps = new Properties();
+            InputStream is = null;
+            try {
+                File props = PropertiesUtil.findConfigFile("server.properties");
+                if (props != null && props.exists()) {
+                    is = new FileInputStream(props);
+                }
+
+                if (is == null) {
+                    is = PropertiesUtil.openStreamFromURL("server.properties");
+                }
+
+                if (is == null) {
+                    System.err.println("Failed to find server.properties");
+                    log.error("Failed to find server.properties");
+                }
+
+                if (is != null) {
+                    serverProps.load(is);
+                }
+
+                EncryptionSecretKeyChecker checker = new EncryptionSecretKeyChecker();
+                checker.check(serverProps);
```

Comment:
_EncryptionSecretKeyChecker.check()_ is with respect to db properties, can define/use other method for server properties (if the encryption type property is different).

Reply:
I've updated the check() to take the argument(encryption type) as an argument - thus making it generic. In terms of the supported types, I think all types supported by db.properties, should be supported for server.properties too. However, I'd like some help wrt testing with encryption type as web